### ISSUE TITLE
Conda Environment File

### ISSUE THREAD

**Issue by pmbaumgartner** [aloctavodia/Statistical-Rethinking-with-Python-and-PyMC3#20]
It might be helpful for contributors to have a [conda environment file](https://conda.io/docs/using/envs.html) so they can quickly create a new environment and get things rolling. I have one I'll paste below, but I haven't checked it against all of the notebooks to make sure everything necessary is included and anything extraneous is not.

Here's mine:

```yaml
name: pymc
channels:
- conda-forge/label/rc
- conda-forge
- defaults
dependencies:
- cycler=0.10.0=py36_0
- freetype=2.7=1
- joblib=0.11=py36_0
- libgpuarray=0.6.2=np112py36_0
- libpng=1.6.28=0
- mako=1.0.6=py36_0
- matplotlib=2.0.0=np112py36_3
- nose=1.3.7=py36_2
- pandas=0.19.2=np112py36_1
- patsy=0.4.1=py36_0
- pyparsing=2.2.0=py36_0
- pytz=2017.2=py36_0
- theano=0.9.0=py36_0
- tqdm=4.11.2=py36_0
- pymc3=3.1rc3=py36_0
- appnope=0.1.0=py36_0
- bleach=1.5.0=py36_0
- decorator=4.0.11=py36_0
- entrypoints=0.2.2=py36_1
- h5py=2.7.0=np112py36_0
- hdf5=1.8.17=1
- html5lib=0.999=py36_0
- ipykernel=4.6.1=py36_0
- ipython=5.3.0=py36_0
- ipython_genutils=0.2.0=py36_0
- jinja2=2.9.6=py36_0
- jsonschema=2.5.1=py36_0
- jupyter_client=5.0.1=py36_0
- jupyter_core=4.3.0=py36_0
- markupsafe=0.23=py36_2
- mistune=0.7.4=py36_0
- mkl=2017.0.1=0
- mkl-service=1.1.2=py36_3
- nbconvert=5.1.1=py36_0
- nbformat=4.3.0=py36_0
- notebook=5.0.0=py36_0
- numpy=1.12.1=py36_0
- openssl=1.0.2k=1
- pandocfilters=1.4.1=py36_0
- path.py=10.1=py36_0
- pexpect=4.2.1=py36_0
- pickleshare=0.7.4=py36_0
- pip=9.0.1=py36_1
- prompt_toolkit=1.0.14=py36_0
- ptyprocess=0.5.1=py36_0
- pygments=2.2.0=py36_0
- python=3.6.1=0
- python-dateutil=2.6.0=py36_0
- pyzmq=16.0.2=py36_0
- readline=6.2=2
- scipy=0.19.0=np112py36_0
- seaborn=0.7.1=py36_0
- setuptools=27.2.0=py36_0
- simplegeneric=0.8.1=py36_1
- six=1.10.0=py36_0
- sqlite=3.13.0=0
- terminado=0.6=py36_0
- testpath=0.3=py36_0
- tk=8.5.18=0
- tornado=4.4.2=py36_0
- traitlets=4.3.2=py36_0
- wcwidth=0.1.7=py36_0
- wheel=0.29.0=py36_0
- xz=5.2.2=1
- zlib=1.2.8=3
- pip:
  - ipython-genutils==0.2.0
  - jupyter-client==5.0.1
  - jupyter-core==4.3.0
  - prompt-toolkit==1.0.14
  - pygpu==0.6.2
prefix: /Users/Peter/anaconda/envs/pymc
```

**Comment by aloctavodia** [comment]
I agree this could be something useful.
Since PyMC3 has several dependencies including NumPy, SciPy, matplotlib. I guess the most minimalistic file could be something like this:

```yml
name: pymc
channels:
- conda-forge
- defaults
dependencies:
  - pymc3
  - jupyter
  - seaborn
  - statsmodels
```
I always install PyMC3 directly from github ```pip install git+https://github.com/pymc-devs/pymc3```. Is this possible using the yml file?

**Comment by pmbaumgartner** [comment]
I tried to include pip git install with a `yml` file earlier this year and never discovered a good solution for it. In my file, I use the `rc` label to install via `conda install -c conda-forge/label/rc pymc3`. I haven't tested this out in a `yml` file though to make sure it gets the version from the `rc` label, I'll give it a shot.

I think a good solution would be to provide the conda file and a pip requirements file, since I know requirements can install direct from github. I can test both solutions and make a PR sometime this week, hopefully.

**Comment by aloctavodia** [comment]
That will be great! @AustinRochford maybe you have some thoughts about this?

**Comment by AustinRochford** [comment]
I think this is a good idea; I don't have a ton of `conda` experience, so I can't comment too deeply on this specific PR.  One thing I will say is that `pycm3` does have a [Dockerfile](https://github.com/pymc-devs/pymc3/blob/master/scripts/Dockerfile) used as a standard environment for testing/development, so that's another option in this direction.

**Comment by aloctavodia** [comment]
fixed  by  #22
